flow.js

```diff
@@ -36,18 +42,27 @@ declare class FusionApp {
   renderer: any;
   enhance<Token, Deps>(
     token: Token,
-    enhancer: (item: Token) => FusionPlugin<Deps, Token>
+    enhancer: (
+      item: $Call<ExtractReturnType, Token>
+    ) => FusionPlugin<Deps, $Call<ExtractReturnType, Token>>
   ): void;
   enhance<Token>(token: Token, enhancer: (token: Token) => Token): void;
   register<Deps, Provides>(Plugin: FusionPlugin<Deps, Provides>): aliaser<*>;
-  register<Deps, Provides>(
-    token: Provides,
-    Plugin: FusionPlugin<Deps, Provides>
+  register<Token, Deps>(
+    token: Token,
+    Plugin: FusionPlugin<Deps, $Call<ExtractReturnType, Token>>
   ): aliaser<*>;
-  register<Token: string>(token: Token, val: string): aliaser<*>;
-  register<Token: number>(token: Token, val: number): aliaser<*>;
-  register<Token: Object>(token: Token, val: $Exact<Token>): aliaser<*>;
-  middleware<Deps>(deps: Deps, middleware: (Deps) => Middleware): void;
+  // register<Token: null>(token: Token, val: null): aliaser<*>;
```

Comment:
Thoughts on no longer supporting arbitrary tokens (i.e. those not created via `createToken`)?  While the freedom is nice, this adds a layer of complexity (more so with optional dependencies) that I'm not convinced is worthwhile.

Reply:
+1

Reply:
One could make an argument for string tokens, but definitely nothing else